### ISSUE TITLE
[destination prefix] Cant add "/" to Destination Stream Prefix

### ISSUE THREAD

**Issue by craunas** [airbytehq/airbyte#35441]
### Topic

"/" seem to be seen as invalid character in Destination Stream Prefix since version 0.50.3 

### Relevant information

In version 0.50.51 it looks like this when trying to add a slash in destination stream prefix:

<img width="550" alt="image" src="https://github.com/airbytehq/airbyte/assets/111738182/71c96c88-1999-4c7e-a470-283a5c9bdf2d">

And like this in version 0.50.3:
<img width="515" alt="image" src="https://github.com/airbytehq/airbyte/assets/111738182/c14710a1-e3fa-4551-99f7-80fd49ac4705">

We've used slashes to extract data into separate directories in object storage (azure blob storage) using the same destination connector.

To achieve the same result now we need to setup multiple destination connector where we specify the directory in the container propery, like "container_name/directory".

It was nice to previolusly have been able to do it in each connection and reusing the destination.

**Comment by marcosmarxm** [comment]
I think this is a design decision because not all databases allow to create tables or objects using `_`. This configuration today uses a generic rule instead of being specific to each destination. For now the workaround is to remove that for another char `container_name_directory`.

**Comment by craunas** [comment]
> I think this is a design decision because not all databases allow to create tables or objects using `_`. This configuration today uses a generic rule instead of being specific to each destination. For now the workaround is to remove that for another char `container_name_directory`.

Ok, yes I thought something like that would be the reason. But it's been a really nice way for us to simplify the setup when we just need to setup one Azure blob storage destination and then decide the path/dir in each connection. Instead of, as now, we have to setup multiple destinations for it. 

> For now the workaround is to remove that for another char `container_name_directory`.

You mean that the solution could be to add a new connection property where the user can specify dir? Yes, that'd work great for our use case.

**Comment by timroes** [comment]
Refinement Notes:

* We need to check here with the destination team what's the supposed way a (file) connectors would handle a `/` as. Is the subdirectory creation just happening by accident, or is this a clear documented feature. If so, how can a connector communicate to us (new metadata?) that it'll support the `/` for the UI to allow it. @nataliekwong to check with team

**Comment by malikdiarra** [comment]
Require more information from the destination team. The metadata should tell us what character are allowed

**Comment by nataliekwong** [comment]
From destinations: IMO it seems like something we should be able to support generally. For S3, GCS and ABS - I think this is a non issue (directories aren't real) for the File system ones (Local JSON, Local CSV) I can't think of a good reason we shouldn't support it

**Comment by octavia-squidington-iii** [comment]
At Airbyte, we seek to be clear about the project priorities and roadmap.  This issue has not had any activity for 180 days, suggesting that it's not as critical as others.  It's possible it has already been fixed. It is being marked as stale and will be closed in 20 days if there is no activity. To keep it open, please comment to let us know why it is important to you and if it is still reproducible on recent versions of Airbyte.

**Comment by nataliekwong** [comment]
Still relevant for file destinations.

**Comment by clnoll** [comment]
This issue has been fixed because we are no longer validating the stream prefix.

In [this PR](https://github.com/airbytehq/airbyte-platform-internal/pull/14503), we removed `SyncCatalog` v1 - related code and obsolete components, including the `DestinationStreamNamesModal`, where the validation that was causing this issue lived.

The stream prefix is now managed in the connection settings, and does not have any validation. This raises the question of whether we should add validation back in; here's the [issue](https://github.com/airbytehq/airbyte/issues/6661#issuecomment-989200860) that motivated it in the first place. If we do add it back in, we should consider separate validation for each connector subtype.

cc @nataliekwong
